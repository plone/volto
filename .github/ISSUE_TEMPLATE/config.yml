blank_issues_enabled: false
contact_links:
  - name: Get support
    url: https://community.plone.org/
    about: Get support for Plone in the Community Forum.
  - name: Chat
    url: https://discord.gg/zFY3EBbjaj
    about: Chat on Discord.
