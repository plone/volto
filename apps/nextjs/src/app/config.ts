const settings = {
  apiPath: process.env.NEXT_PUBLIC_VERCEL_URL
    ? // Vercel does not prepend the schema to the NEXT_PUBLIC_VERCEL_URL automatic env var
      `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`
    : 'http://localhost:3000',
};

const config = {
  settings,
};

export default config;
