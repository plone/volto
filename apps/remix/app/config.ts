const settings = {
  apiPath: "http://localhost:8080/Plone",
};

const config = {
  settings,
};

export default config;
