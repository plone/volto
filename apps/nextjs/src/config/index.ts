import installPlone from '@plone/nextjs';
import config from '@plone/registry';

export default installPlone(config);
