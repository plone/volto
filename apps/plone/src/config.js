/** Volto Project Configuration
 * The recommended way for configuring a Volto project is using an add-on.
 * The project should only be considered as boilerplate, and all the code and
 * configuration should happen and be placed in add-ons.
 *
 * Both configuring directly and placing code in a project is discouraged, and might be
 * removed at some point from Volto.
 *
 * The local project is left for backwards compatibility for existing projects.
 */

// [Internal] All the imports of modules required for the configuration *must* happen
// here BEFORE the following line
import '@plone/volto/config';

export default function applyConfig(config) {
  return config;
}
