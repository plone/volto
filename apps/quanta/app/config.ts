/**
 * This is the client side config entry point
 */
import config from '@plone/registry';
// eslint-disable-next-line import/no-unresolved
import applyAddonConfiguration from '../registry.loader';

export default function install() {
  applyAddonConfiguration(config);
  return config;
}
