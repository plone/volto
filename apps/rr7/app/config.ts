/**
 * This is the client side config entry point
 */
import config from '@plone/registry';
import applyAddonConfiguration from '@plone/registry/addons-loader';

export default function install() {
  applyAddonConfiguration(config);
  config.settings.apiPath = 'http://localhost:3000';
  return config;
}
