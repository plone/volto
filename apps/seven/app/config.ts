/**
 * This is the client side config entry point
 */
import config from '@plone/registry';
// eslint-disable-next-line import/no-unresolved
import applyAddonConfiguration from '../.plone/registry.loader';

export default function install() {
  config.settings.defaultLanguage = 'en';
  config.settings.supportedLanguages = ['en'];

  applyAddonConfiguration(config);

  return config;
}
