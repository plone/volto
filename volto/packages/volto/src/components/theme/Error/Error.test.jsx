import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';
import config from '@plone/volto/registry';

import Error from './Error';

beforeAll(() => {
  config.views.errorViews = {
    401: () => <div className="401" />,
  };
});

const mockStore = configureStore();

describe('Generic Error', () => {
  it('renders a not found component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
      apierror: {
        message: 'You are not authorized to access this resource',
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Error error={{ status: 401, message: 'Unauthorized' }} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
