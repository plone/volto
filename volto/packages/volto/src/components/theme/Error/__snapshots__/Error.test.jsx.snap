// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Error renders a not found component 1`] = `
<div
  id="view"
>
  <div
    className="401"
  />
</div>
`;
